astroquery/cadc/core.py

```
@@ -0,0 +1,193 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+=============
+Cadc TAP plus
+=============
+
+"""
+
+from astroquery.cadc.tap import TapPlus
+
+__all__ = ['Cadc', 'CadcTAP']
+
+
+class CadcTAP(object):
+    """
+    Proxy class to default TapPlus object (pointing to CADA Archive)
+    """
+    def __init__(self, url=None, tap_plus_handler=None, verbose=False):
+        """
+        Initialize CadcTAP object
+
+        Parameters
+        ----------
+        url : str, optional, default 'None;
+            a url to use instead of the default
+        tap_plus_handler : TAP/TAP+ object, optional, default 'None'
+            connection to use instead of the default one created
+        verbose : bool, optional, default 'False'
+            flag to display information about the process
+
+        Returns
+        -------
+        CadcTAP object
+        """
+        if tap_plus_handler is None:
+            if url is None:
+                self.__cadctap = TapPlus(
+                    url="http://www.cadc-ccda.hia-iha.nrc-cnrc.gc.ca/tap",
+                    verbose=verbose)
+            else:
+                self.__cadctap = TapPlus(url=url, verbose=verbose)
+        else:
+            self.__cadctap = tap_plus_handler
```

Maybe raise an exception if url is set but, in this case, ignored?

---

I added a raise error if both are sent